### PR TITLE
Separate development and production entries

### LINKED ISSUE #79: console.warn in production
There are a lot of warnings emitted from [`drag-drop-context.tsx`](https://github.com/thisbeyond/solid-dnd/blob/b5d9b4f7d1f4ede56451d6b8990d2df7ec301006/src/drag-drop-context.tsx#L191) which should probably be removed in production build as they are not useful for end users and bloat the bundle.

I've recently made [tsup-preset-solid](https://github.com/solidjs-community/tsup-preset-solid) to help with that so this should be an easy fix

wanna PR? :)

### PR DESCRIPTION
**Pull request by thetarnav**
Fixes #79 

Uses tsup and tsup-preset-solid for building library.
It generates dev entry with preserved console.logs and index (production) with console calls removed.

### CODE REVIEW

File: src/create-draggable.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ const createDraggable = (id: Id, data: Record<string, any> = {}): Draggable => {
 
           if (!transformsAreEqual(resolvedTransform, noopTransform())) {
             const style = transformStyle(transform());
-            element.style.setProperty("transform", style.transform);
+            element.style.setProperty("transform", style.transform ?? null);
```
Comment (thetarnav) [inline src/create-draggable.ts:81]: these were causing ts errors in build (`undefined is not assignable to string | null`)

File: tsup.config.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { defineConfig } from "tsup-preset-solid";
+
+export default defineConfig(
+  {
+    entry: "src/index.tsx",
+    devEntry: true,
+  },
+  {
+    // Uncomment this to update package.json
```
Comment (martinpengellyphillips) [inline tsup.config.ts:9]: Curious: What is the use case for this? Why leave it there commented out?

Comment (thetarnav) [reply tsup.config.ts:9]: so I left it commented out mostly for your information that this is a thing 🤷‍♂️

**Comment by martinpengellyphillips** [comment]
I'll aim to test and merge this by end of week.

Out of interest @thetarnav - will this also resolve this class of issue that keeps coming up for folks: #78

**Comment by thetarnav** [comment]
What can I do to reproduce this issue (haven't seen it before)?

**Comment by martinpengellyphillips** [comment]
Someone shared this repro https://stackblitz.com/edit/github-sadxvo?file=package.json - look in the console to see error. The general fix is to add the following to vite config, but I'm curious if there is something we can do in this package to mitigate.

```
  optimizeDeps: {
    extensions: ['jsx'],
  },
```

**Comment by thetarnav** [comment]
Ok I think it does resolve it
Downloaded the project - the same error
Linked the newly built package - no error
I don't understand what exactly is the issue, but I expect it had something to do with mixed .js and .jsx files in the /source output. Now there is a single file for each entry so everything is transformed equally.

**Comment by martinpengellyphillips** [comment]
Ah, that's great - it's the most frequent stumbling block for folks so good to get it solved. Thanks for testing!

**Comment by martinpengellyphillips** [comment]
Thanks again for this - released now.

**Pull request merged**